src/posix/platform/logging.c

```diff
@@ -79,9 +78,33 @@ OT_TOOL_WEAK void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const
     otEXPECT_ACTION(charsWritten >= 0, logString[offset] = 0);
 
 exit:
-    syslog(LOG_CRIT, "%s", logString);
+    switch (aLogLevel)
+    {
+    case OT_LOG_LEVEL_NONE:
+        logLevel = LOG_ALERT;
+        break;
+    case OT_LOG_LEVEL_CRIT:
+        logLevel = LOG_CRIT;
+        break;
+    case OT_LOG_LEVEL_WARN:
+        logLevel = LOG_WARNING;
+        break;
+    case OT_LOG_LEVEL_NOTE:
+        logLevel = LOG_NOTICE;
+        break;
+    case OT_LOG_LEVEL_INFO:
+        logLevel = LOG_INFO;
+        break;
+    case OT_LOG_LEVEL_DEBG:
+        logLevel = LOG_DEBUG;
+        break;
+    default:
+        assert(false);
+        logLevel = LOG_DEBUG;
+        break;
+    }
+    syslog(logLevel, "%s", logString);
 
```

Comment:
The `platformLoggingInit()` sets the syslog mask to `setlogmask(setlogmask(0) & LOG_UPTO(LOG_NOTICE));`. Shall we change this to be `LOG_DEBUG` with this change?